src/storage.rs

```diff
@@ -56,9 +56,44 @@ impl RaftState {
     }
 }
 
+/// Purpose of getting entries.
+#[derive(Debug)]
+pub enum GetEntriesFor {
```

Comment:
Why need to make it public?

Reply:
In `fn on_raft_log_fetched`, we need to do different things according to the different type of `GetEntriesFor`. If `GetEntriesFor` is `SendAppend`, the peer must be the leader. But if `GetEntriesFor` is `SendForward`, the peer should not be the leader.

Reply:
> If GetEntriesFor is SendAppend, the peer must be the leader. But if GetEntriesFor is SendForward, the peer should not be the leader.

Why? What if the role is changed when the logs are being fetched?

Reply:
The agent is the follower. So we need to check whether it is still the follower but not the leader in `SendForward` situation. In addition, we check the term in `fn on_raft_log_fetched`. If the role is changed, the term will be changed too.

Reply:
Does it matter? What could go wrong if always update the fetched result and clear it afterwards?

Reply:
The peer selected as agent must be a follower. In the original `fn on_raft_log_fetched`, it first checks whether this peer is the leader. If this peer is not the leader now, it will return. Hence, if the async fetch result is temporarily unavailable, the callback function will never be called  successfully for the agent.
However, it does not affect the correctness of follower replication. It is just like that the agent always fetches log entries in sync way.